extension-style/src/main/java/com/mapbox/maps/extension/style/layers/properties/generated/Property.kt

```diff
@@ -533,6 +533,46 @@ enum class Anchor(override val value: String) : LayerProperty {
    */
   VIEWPORT("viewport"),
 }
+// NAME: The name of the projection to be used for rendering the map.
+/**
+ * The name of the projection to be used for rendering the map.
+ *
+ * @param value String value of this property
+ */
+enum class Name(override val value: String) : LayerProperty {
+  /**
+   * An Albers equal-area projection centered on the continental United States. You can configure the projection for a different region by setting {@link CENTER} and {@link PARALLELS} properties. You may want to set max bounds to constrain the map to the relevant region.
+   */
+  ALBERS("albers"),
+  /**
+   * An Equal Earth projection.
+   */
+  EQUALEARTH("equalEarth"),
+  /**
+   * An Equirectangular projection. This projection is very similar to the Plate Carrée projection.
+   */
+  EQUIRECTANGULAR("equirectangular"),
+  /**
+   * A Lambert conformal conic projection. You can configure the projection for a region by setting {@link CENTER} and {@link PARALLELS} properties. You may want to set max bounds to constrain the map to the relevant region.
+   */
+  LAMBERTCONFORMALCONIC("lambertConformalConic"),
+  /**
+   * The Mercator projection is the default projection.
+   */
+  MERCATOR("mercator"),
+  /**
+   * A Natural Earth projection.
+   */
+  NATURALEARTH("naturalEarth"),
+  /**
+   * A Winkel Tripel projection.
+   */
+  WINKELTRIPEL("winkelTripel"),
+  /**
+   * A globe projection.
+   */
+  GLOBE("globe"),
+}
```

Comment:
@kiryldz does gl native support all of these? I thought only globe and mercator were supported so far.

Reply:
I did add the test covering this, please see actual instrument tests in `StyleProjectionTest` file.

Reply:
can we use `_` in the names?. for eg. `LAMBERT_CONFORMAL_CONIC` instead of `LAMBERTCONFORMALCONIC`.

Reply:
@ank27 fixed!

Reply:
I think putting these constants in the public API will suggest to developers that they can actually be used and will result in avoidable support burden.

Reply:
we can remove these unsupported enums from the style-parser.js for now, and add them back when they become available to gl-native. The only concern is that adding something to enum later would be breaking change.

Reply:
@pengdev I did remove not supported enums. Good point about breaking change, let me ask if gl-native is planning to support other projections in nearest future.